ext/ensemble_builder.jl

```diff
@@ -0,0 +1,364 @@
+import ClimaCalibrate.EnsembleBuilder as EnsembleBuilder
+import ClimaCalibrate: g_ens_mat
+
+"""
+    MetadataInfo{METADATA <: Metadata}
+
+An object that stores the metadata of the observation, the index of the
+metadata, and the corresponding range to fill out in the G ensemble matrix.
+"""
+struct MetadataInfo{METADATA <: Metadata}
+    """The index of the metadata of all metadata"""
+    index::Int64
+
+    """The indices of column of the current G ensemble matrix to fill out"""
+    range::UnitRange{Int64}
+
+    """A single metadata from the metadata for the current minibatch"""
+    metadata::METADATA
+end
+
+"""
+    GEnsembleBuilder{FT <: AbstractFloat}
+
+An object to help build G ensemble matrix by using the metadata stored in the
+`EKP.EnsembleKalmanProcess` object. Metadata must come from `ClimaAnalysis`.
+
+`GEnsembleBuilder` takes in preprocessed `OutputVar`s and automatically
+construct the corresponding G ensembe matrix for the current iteration of the
+calibration.
+"""
+struct GEnsembleBuilder{FT <: AbstractFloat, METADATAINFO <: MetadataInfo}
```

Comment:
It would be nice to add a summary or show method for this struct just to show the size, short names of vars in one column and what percent are completed.